connectors/anaplan/app/handlers.py

```
@@ -0,0 +1,234 @@
+                }
+                metadata_items.append(app_item)
+                active_apps.append((app_guid, app_name))
+
+            # Step 3: Get all pages for all apps concurrently
+            logger.info("Fetching pages for all apps...")
+
+            # Get all pages for the active apps
+            all_pages = await extract_pages_with_details(
+                self.client, {app[0] for app in active_apps}
+            )
+
+            # Group pages by app GUID
+            app_pages_mapping = {}
+            for page in all_pages:
+                app_guid = page.get("appGuid")
+                if app_guid and app_guid in {app[0] for app in active_apps}:
+                    if app_guid not in app_pages_mapping:
+                        app_pages_mapping[app_guid] = []
+                    app_pages_mapping[app_guid].append(page)
+
+            # Add page items to app children
+            for app_guid, app_name in active_apps:
+                pages_for_app = app_pages_mapping.get(app_guid, [])
+
+                # Find the app item and add pages to its children
+                for app_item in metadata_items:
+                    if app_item["value"] == app_guid:
+                        for page in pages_for_app:
+                            page_item = {
+                                "value": page["guid"],
+                                "title": page["name"],
+                                "children": [],
+                            }
+                            app_item["children"].append(page_item)
+                        break
+
+            return metadata_items  # Return list of nested metadata items
+
+        except Exception as e:
+            logger.error(f"Failed to fetch metadata: {str(e)}")
+            # For errors, we should raise an exception rather than return a list
+            # The SDK will handle the exception and return appropriate error response
+            raise Exception(f"Failed to fetch metadata: {str(e)}")
```

I think you should raise a specific exception like `application_sdk.common.error_codes.ClientError` not `Exception`